### PR TITLE
MueLu: add greedy Dirichlet option

### PR DESCRIPTION
**Pull request by pohm01**
Add and option to treat a node as Dirichlet if any of the DOFs on that node are Dirichlet (the default behavior is that nodes are marked as Dirichlet only if all DOFs on the node are Dirichlet).

@trilinos/muelu 

## Motivation
Observed convergence benefits in some Drekar simulations rppawlo/DrekarBase#546. Further investigation could be useful.

### CODE REVIEW

**Review by mayrmt — COMMENTED** [review]
@pohm01 I like this new feature! Looks good, but could you please add a test for the new feature?

File: packages/muelu/doc/UsersGuide/masterList.xml [hunk 1]
```diff
@@ -608,6 +608,14 @@
       <comment-ML>parameter not existing in ML</comment-ML>
     </parameter>
 
+    <parameter>
+      <name>aggregation: greedy Dirichlet</name>
+      <type>bool</type>
+      <default>false</default>
+      <description>Force the node to be Dirichlet if any DOFs on the node are Dirichlet (default is nodes are Dirichlet only if all DOFs on node are Dirichlet).</description>
```
Comment (jhux2) [inline packages/muelu/doc/UsersGuide/masterList.xml:615]: Nitpick -- someone might confuse the use of node here with compute node, when what's really meant is aggregate.

**Review by jhux2 — CHANGES_REQUESTED** [review]
@mayrmt is right, could you please add a unit test?

**Review by jhux2 — APPROVED** [review]
lgtm, thanks!

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**